python/pyarrow/_parquet.pyx

```diff
@@ -287,6 +289,17 @@ cdef class RowGroupMetaData:
         return self.metadata.total_byte_size()
 
 
+def _reconstruct_filemetadata(Buffer serialized):
+    cdef:
+        FileMetaData metadata = FileMetaData.__new__(FileMetaData)
+        CBuffer *buffer = serialized.buffer.get()
+        uint32_t metadata_len = buffer.size()
```

Comment:
I think this is the line that causes the AppVeyor compile error. You need an explicit cast to `uint32_t` to avoid the implicit truncation.